### PR TITLE
store/consul: Add MutatePod

### PR DESCRIPTION
**Pull request by petertseng**
This allows for transactionally transforming a number of pod manifests
in some way. A first use case is to replace a signature.

### CODE REVIEW

File: pkg/store/consul/kv.go [hunk 1]
```diff
@@ -226,6 +228,59 @@ func (c consulStore) DeletePod(podPrefix PodPrefix, nodename types.NodeName, pod
 	return writeMeta.RequestTime, nil
 }
 
+// TransformPod mutates the input context in such a way
+// that it will transactionally transform the pod manifest on the given nodes.
+// If the pod does not exist on some of the given nodes,
+// the operation skips those nodes without considering that an error.
+func (c consulStore) TransformPod(
```
Comment (mpuncel) [inline pkg/store/consul/kv.go]: nit: we've used the word `Mutate` elsewhere e.g. https://github.com/square/p2/blob/master/pkg/store/consul/pcstore/consul_store.go#L230

**Comment by rudle** [comment]
Sweet! How did you test this?

**Comment by petertseng** [comment]
Now called Mutate and with included test

**Comment by petertseng** [comment]
ds test flaked on me:

```
--- FAIL: TestPublishToReplication (1.03s)
	daemon_set_test.go:501: Unexpectedly no error when no nodes are eligible
```

But it's OK, that was my first commit not my second. So I can still merge.

**Pull request merged**